src/miniscript/mod.rs

```diff
@@ -59,7 +59,7 @@ pub struct Miniscript<Pk: MiniscriptKey, Ctx: ScriptContext> {
     pub ty: types::Type,
     ///Additional information helpful for extra analysis.
     pub ext: types::extra_props::ExtData,
-    /// Context PhantomData. Only accessible inside this crate
+    /// Needed because not all terminals are paramatized by script context.
```

Comment:
In c874074b4a6dc6c7e47be1a8945b0a6aa015e437

This is actually just used to simulate `#[non_exhaustive]` which we did not have in 1.29. We can just drop it and add `#[non_exhaustive]`.

Reply:
Did you think this was an enum? I think the comment is correct IIUC, if we remove the phantom we get build error
```
error[E0392]: parameter `Ctx` is never used
  --> src/miniscript/mod.rs:55:42
   |
55 | pub struct Miniscript<Pk: MiniscriptKey, Ctx: ScriptContext> {
   |                                          ^^^ unused parameter
   |
   = help: consider removing `Ctx`, referring to it in a field, or using a marker such as `PhantomData`
```

Reply:
Oh, no, I think actually I'm just wrong. But I don't understand your comment. `Ctx` is clearly used in the `node` field, this error is just a bug in rustc (which I think has been well-known for many years). We should change the comment to clarify this.

And it *is* doing double-duty as a `#[non_exhaustive]` tag, which is fine. It prevents users from directly constructing this type.

Reply:
Oh I thought it was being caused by the fact that not all variants of `Terminal` have `Ctx`. Not sure what to write there now, the original is definitely just noise though.